archive/vcs.go

```diff
@@ -33,7 +39,44 @@ var VCSList = []*VCS{
 	&VCS{
 		Name:   "git",
 		Detect: []string{".git/"},
-		Files:  vcsFilesCmd("git", "ls-files"),
+		Preflight: func(path string) error {
```

Comment:
I recommend putting this into its own function just for readability reasons (to keep the struct definition small)